OptimizelySDKCore/OptimizelySDKCoreTests/OPTLYNotificationCenterTest.m

```diff
@@ -0,0 +1,211 @@
+    XCTAssertEqual(1, notificationId);
+    XCTAssertEqual(1, _notificationCenter.notificationsCount);
+    
+    // Verify that callback removed successfully.
+    XCTAssertEqual(YES, [_notificationCenter removeNotification:notificationId]);
+    XCTAssertEqual(0, _notificationCenter.notificationsCount);
+    
+    //Verify return false with invalid ID.
+    XCTAssertEqual(NO, [_notificationCenter removeNotification:notificationId]);
+    
+    // Verify that callback added successfully and return right notification ID.
+    XCTAssertEqual(_notificationCenter.notificationId, [_notificationCenter addNotification:OPTLYNotificationTypeActivate activateListener:_activateNotification]);
+    XCTAssertEqual(1, _notificationCenter.notificationsCount);
+}
+
+- (void)testAddSameNotificationListenerMultipleTimes {
+    [_notificationCenter addNotification:OPTLYNotificationTypeActivate activateListener:_activateNotification];
+    
+    // Verify that adding same callback multiple times will gets failed.
+    XCTAssertEqual(-1, [_notificationCenter addNotification:OPTLYNotificationTypeActivate activateListener:_activateNotification]);
+    XCTAssertEqual(1, _notificationCenter.notificationsCount);
+}
+
+- (void)testAddInvalidNotificationListeners {
+    // Verify that AddNotification gets failed on adding invalid notification listeners.
+    XCTAssertEqual(0, [_notificationCenter addNotification:OPTLYNotificationTypeTrack activateListener:_activateNotification]);
+    XCTAssertEqual(0, [_notificationCenter addNotification:OPTLYNotificationTypeActivate trackListener:_trackNotification]);
+    
+    // Verify that no notifion has been added.
+    XCTAssertEqual(0, _notificationCenter.notificationsCount);
+}
+
+- (void)testClearNotifications {
+
+    // Add activate notifications.
+    [_notificationCenter addNotification:OPTLYNotificationTypeActivate activateListener:_activateNotification];
+    [_notificationCenter addNotification:OPTLYNotificationTypeActivate activateListener:_anotherActivateNotification];
+    
+    // Add track notification.
+    [_notificationCenter addNotification:OPTLYNotificationTypeTrack trackListener:_trackNotification];
+    
+    // Verify that callbacks added successfully.
+    XCTAssertEqual(3, _notificationCenter.notificationsCount);
+    
+    // Verify that only decision callbacks are removed.
+    [_notificationCenter clearNotifications:OPTLYNotificationTypeActivate];
+    XCTAssertEqual(1, _notificationCenter.notificationsCount);
+    
+    // Verify that ClearNotifications does not break on calling twice for same type.
+    [_notificationCenter clearNotifications:OPTLYNotificationTypeActivate];
+    [_notificationCenter clearNotifications:OPTLYNotificationTypeActivate];
+    
+    // Verify that ClearNotifications does not break after calling ClearAllNotifications.
+    [_notificationCenter clearAllNotifications];
```

Comment:
Some code comments mention "ClearAllNotifications", but should say "clearAllNotifications"
with lower case "c" .